lib/routes/icable/news.ts

```diff
@@ -0,0 +1,243 @@
+}
+
+async function handler(ctx) {
+    const category = ctx.req.param('category') ?? '新聞資訊';
+    const limit = ctx.req.query('limit') ?? 20;
+    const embed = ctx.req.query('embed') === '1';
+    const graphqlUrl = 'https://www.i-cable.com/graphql';
+
+    const getMainCategoryResponse = await cache.tryGet(
+        'icables-mainCategories',
+        async () =>
+            await got(graphqlUrl, {
+                method: 'POST',
+                body: getMainCategoryQuery(),
+            }),
+        config.cache.routeExpire,
+        false
+    );
+
+    const mainCategories = getMainCategoryResponse.data.data.mainCategory.map((category) => ({
+        name: category.name,
+        id: category.id,
+    }));
+
+    const getCategoriesResponse = await Promise.all(
+        mainCategories.map(
+            (mainCategory) =>
+                cache.tryGet(
+                    `icables-categories-${mainCategory.id}`,
+                    async () =>
+                        await got(graphqlUrl, {
+                            method: 'POST',
+                            body: getCategoryEdgesQuery(mainCategory.id),
+                        })
+                ),
+            config.cache.routeExpire,
+            false
+        )
+    );
+
+    const categories = getCategoriesResponse.map((response) =>
+        response.data.data.categories.edges.map((category) => ({
+            name: category.node.name,
+            id: category.node.databaseId,
+        }))
+    );
+
+    const categoriesIds = [...mainCategories, ...categories.flat()];
+
+    const categoryInfo = categoriesIds.find((categoryA) => categoryA.name === category || categoryA.id === +category);
+
+    const getListsByCateResponse = await got(graphqlUrl, {
+        method: 'POST',
+        body: getListsByCateQuery(categoryInfo.id, limit),
+    });
+
+    const list = getListsByCateResponse.data.data.posts.edges.map((value) => {
+        const { id, title, link, date, categories, featuredImage } = value.node;
+        const category = categories.edges.map((node) => node.node.name);
+        const pubDate = timezone(parseDate(date), +8);
+        const image = featuredImage ? featuredImage.node : [];
+        return {
+            id,
+            title,
+            link,
+            image,
+            pubDate,
+            category,
+        };
+    });
+
+    const feed = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: item.link,
+                });
+                const $ = load(detailResponse.data);
+                item.article = $('article.post-detail-wrap > p')
+                    .map((i, el) => $(el).html())
+                    .toArray();
+                item.video = $('article.post-detail-wrap video')
+                    .map((i, el) => ({
+                        sourceUrl: $(el).attr('data-url'),
```

Comment:
Maybe I've missed something here, but I don't think placing m3u8 link in source will make it playable without aids from external JS lib.